### ISSUE TITLE
Definition of primitive type

### ISSUE THREAD

**Issue by robertlarkins** [dotnet/docs#7792]
This page covers the `IsPrimitive` property for types, but I haven't been able to find a definition anywhere of what a primitive type is. As far as I can tell primitive types are the value types built into .Net. Structs and Enums aren't primitive types (but are value types) as they can be user defined.

There was this [discussion](https://stackoverflow.com/questions/2066035/in-c-sharp-are-the-terms-primitive-and-literal-interchangeable/2135136#2135136), that says that 'primitive' isn't defined in the specification, but there must be a reason why the types listed on this page are deemed primitive.

This [site](https://medium.com/omarelgabrys-blog/primitive-data-types-in-c-vs-java-5b8a597eef05) describes primitives as the main built-in types, but `string`, `object`, aren't primitive as per the list on the `IsPrimitive` page, but maybe this definition isn't specific to value types.

Is there a definition, and can it be added to this page, and maybe else where in the msdn documentation?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: dffb41b0-35d3-6ddc-7bfe-c162d1e32d45
* Version Independent ID: 51569306-3108-5d27-466e-44aa6acb3514
* Content: [Type.IsPrimitive Property (System)](https://docs.microsoft.com/en-us/dotnet/api/system.type.isprimitive?view=netframework-4.7.2)
* Content Source: [xml/System/Type.xml](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System/Type.xml)
* Product: **dotnet-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by rpetrusha** [comment]
Thanks for pointing out our failure to define primitive, @robertlarkins. 

A simple type (in C#) is the same as a primitive type (in .NET). While the CLI specification does not define *primitive type*, the C# specification does define *simple type*. A simple type is one of a predefined set of struct types that are identified by the compiler through reserved words. That is, a simple type is predefined type recognized by the language compiler that corresponds directly to a .NET type. 

Object isn't a primitive type since it's a reference type. String isn't a primitive type both because it consists of cone or more Char instances (which are a primitive type) and because it's a reference type.

We'll add this issue to the backlog as well as to the Community Contributor's project,

**Comment by robertlarkins** [comment]
Thanks for the description. By this definition is an `enum` not a simple type because you declare what types their members are (the default being int)?

Also, from the spec (9.3.5) it says

> Most simple types permit values to be created by writing literals

which of the listed simple types can't be assigned by a literal?

**Comment by rpetrusha** [comment]
Yes. An enum is not a primitive, since it has multiple fields that themselves are primitives.

The values of al the simple types in both C# and Visual Basic can be defined by using literal values, in a few cases along with a type identifier, as in `float value = 12.3f;` or with a special syntax, as in `char ch = 'A';`
